File: leetcode/121BestTimeToBuyAndSellStock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = Integer.MAX_VALUE;
        int ans = 0;
        
        for(int i=0;i<prices.length;i++){
            if(min > prices[i])
                min = prices[i];
            
            if(ans < (prices[i]-min))
                ans = prices[i]-min;
        }
        
        return ans;
    }
}

File: leetcode/121BestTimeToBuyAndSellStock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = float('-inf')
        min_price = float('inf')
        for price in prices:
            min_price = min(min_price, price)
            profit = price - min_price
            max_profit = max(max_profit, profit)
        return max_profit
